app/src/main/java/edu/uc/willi6jd/mycaloriecounter/FoodActivity.kt

```diff
@@ -43,7 +44,7 @@ class FoodActivity : AppCompatActivity() {
             val foodItemArray = Array(20) { }
 
             // Push foodItem to FoodItem Array
-            foodItemArray.add(foodItem(nm, cal, sgr, pro))
+            foodItemArray.add(foodItem(itemName, itemCalories, itemSugar, itemProtein))
```

Comment:
This looks better, as the code is more self-descriptive.  But... it looks like you are using the Views from the layout, where the previous version of function call this used parameters from the foodItem function.  So, different (and incompatible) types.  Do you need to alter the function parameter names to match these more descriptive names?  If not, can the function parameters be removed entirely, and you invoke .text.toString on the View objects?